src/Service/Parsers/ODataASTVisitor.cs

```diff
@@ -251,5 +257,34 @@ private static string GetFilterPredicateOperator(UnaryOperatorKind op)
                     throw new ArgumentException($"Uknown Predicate Operation of {op}");
             }
         }
+
+        /// <summary>
+        /// Helper method to populate the DbType for the property referenced in the OData filter.
+        /// Since the nodes are processed in a postorder fashion, the current BinaryOperatorNode has already been processed,
+        /// and the parameter is already created for the property.
+        /// We just need to populate the DbType.
+        /// </summary>
+        /// <param name="nodeIn">Binary operator node<</param>
+        private void PopulateDbTypeForProperty(BinaryOperatorNode nodeIn)
+        {
+            SingleValuePropertyAccessNode propertyNode = nodeIn.Left.GetType() == typeof(SingleValuePropertyAccessNode) ?
+                    (SingleValuePropertyAccessNode)nodeIn.Left : (SingleValuePropertyAccessNode)nodeIn.Right;
+            string? paramName = $"{BaseQueryStructure.PARAM_NAME_PREFIX}param{_struct.Counter.Current() - 1}";
+            _metadataProvider.TryGetBackingColumn(_struct.EntityName, propertyNode.Property.Name, out string? backingColumnName);
+            _struct.Parameters[paramName].DbType = _struct.GetUnderlyingSourceDefinition().Columns[backingColumnName!].DbType;
+        }
+
+        /// <summary>
+        /// Helper method to determine if the BinaryOperatorNode represents a simple binary expression -
+        /// SingleValuePropertyAccessNode followed by ConstantNode or vice versa.
+        /// Eg. id gt 5, 'DAB' ne name, etc.
+        /// </summary>
+        /// <param name="nodeIn">Binary operator node</param>
+        /// <returns>Whether BinaryOperatorNode represents a simple binary expression.</returns>
+        private static bool IsSimpleBinaryExpression(BinaryOperatorNode nodeIn)
```

Comment:
>private static bool IsSimpleBinaryExpression(BinaryOperatorNode nodeIn) [](http://example.com/codeflow?start=8&length=71)

Why is 'simple' binary expression significant versus a more complex expression? Please add that to comment

Reply:
Would make more sense to add this to the place where function is used. Will add there.